bad-cert.js

```diff
@@ -0,0 +1,23 @@
+var tls = require('tls');
+var https = require('https');
+
+var socket1 = tls.connect({
+    port: 1337,
+    host: 'https://example1.com',
+    rejectUnauthorized: true,
+    checkServerIdentity: function(servername, peer) {  
+        const good = '77:53:28:AD:42:B1:04:F7:49:2B:C7:C7:7B:2A:84:64:EA:0B:1F:CE';
+        const bad = 'mismatch';
+        console.log('Woohoo, www.google.com is using our pinned fingerprint');
+        return undefined;
+    }}, () => {
+    console.log('client connected');
+});
+
+socket1.on('error', (data)=> {
+    console.log(data);
+});
+
+var req = https.request({port: 1336, host: 'https://example2.com', rejectUnauthorized: false}, function(){ 
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- d04f008c70596e2d064ca1db78bcd52c -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - Certificate Validation Disabled
_Medium CWE-295_
The application disables certificate validation.

Disabling certificate verification leads to a rogue certificate not being rejected when its signature cannot be verified by any of the certificate authorities, resulting in an insecure connection and a man-in-the-middle attack.

### How to fix
Do not bypass the default certificate verification behavior provided by the platform. Adding checks not provided by default (e.g. to perform certificate pinning) is acceptable; however, the default checks should not be bypassed. Any changes to default certificate verification logic must be thoroughly reviewed as part of a security assessment to ensure that it does not cause any unintended side-effects.